### DISCUSSION TITLE
proc macro like in rust?

### DISCUSSION THREAD

**Discussion by felixf4xu** [vlang/v#19496] [Ideas]
HI,

I like vlang's simplicity, and I also like rust's proc macro (https://doc.rust-lang.org/reference/procedural-macros.html), which really makes rust unique.

I'd like to check if it can be supported in vlang?

**Comment by JalonSolov** [comment]
V doesn't support macros, and hopefully never will.

However, at least some of the functionality mentioned may be achievable in V without resorting to macros.  With compile-time functions, run-time reflection, hot-code reloading, etc., you can do all sorts of fun things.

**Reply by jacksonmowry** [reply]
Please no macros
